index.bs

```diff
@@ -328,6 +331,56 @@ When this method is invoked, the user agent MUST run the following steps:
 
 </div>
 
+<div class="algorithm" data-algorithm="fill-joint-radii">
+
+The <dfn method for="XRFrame">fillJointRadii(sequence&lt;XRJointSpace&gt; |jointSpaces|, Float32Array |radii|)</dfn> method populates |radii| with the radii of the |jointSpaces|.
+
+When this method is invoked on an {{XRFrame}} |frame|, the user agent MUST run the following steps:
+
+  1. Let |frame| be [=this=].
+  1. Let |session| be |frame|'s {{XRFrame/session}} object.
+  1. If |frame|'s [=active=] boolean is <code>false</code>, throw an {{InvalidStateError}} and abort these steps.
+  1. For each |joint| in the |jointSpaces|:
+    1. If |joint|'s [=XRSpace/session=] is different from |session|, throw an {{InvalidStateError}} and abort these steps.
+  1. If the length of |jointSpaces| is larger than the number of elements in |radii|, throw a {{DataError}} and abort these steps.
+  1. let |offset| be a new number with the initial value of <code>0</code>.
+  1. For each |joint| in the |jointSpaces|:
+    1. Set the float value of |radii| at |offset| to <code>NaN</code>.
+    1. If the user agent can determine the [=skeleton joint/radius=] of |joint|, set the float value of |radii| at |offset| to that [=skeleton joint/radius=].
+    1. Increase |offset| by <code>1</code>.
+
+</div>
+
+<div class="algorithm" data-algorithm="fill-poses">
+
+The <dfn method for="XRFrame">fillPoses(sequence&lt;XRSpace&gt; |spaces|, XRSpace |baseSpace|, Float32Array |transforms|)</dfn> method populates |transforms|
+with the matrices of the poses of the |spaces| relative to the |baseSpace|.
+
+When this method is invoked on an {{XRFrame}} |frame|, the user agent MUST run the following steps:
+
+  1. Let |frame| be [=this=].
+  1. Let |session| be |frame|'s {{XRFrame/session}} object.
+  1. If |frame|'s [=active=] boolean is <code>false</code>, throw an {{InvalidStateError}} and abort these steps.
+  1. For each |space| in the |spaces| sequence:
+    1. If |space|'s [=XRSpace/session=] is different from |session|, throw an {{InvalidStateError}} and abort these steps.
+  1. If |baseSpace|'s [=XRSpace/session=] is different from |session|, throw an {{InvalidStateError}} and abort these steps.
+  1. If the length of |spaces| multiplied by <code>16</code> is larger than the number of elements in |transforms|, throw a {{DataError}} and abort these steps.
+  1. let |offset| be a new number with the initial value of <code>0</code>.
+  1. Initialize |pose| as follows:
```

Comment:
I was hoping that as well but then I can't fall back on the algorithm in the WebXR spec.

Reply:
I agree that it would be good to allow this language to avoid the "create a pose" wording, but I think it's fine for now. At the very least it only creates one for the whole list of spaces. Plus, I would expect that browsers could silently avoid actually creating one in favor of just performing the same math and they'd be OK.

Put it in as-is and file an issue to improve later, I say.